packages/components/pie-modal/src/index.ts

```diff
@@ -56,39 +122,19 @@ export class PieModal extends RtlMixin(LitElement) {
         `;
     }
 
-    /**
-     * We require this because toggling the prop `isOpen` itself won't
-     * allow the dialog to open in the correct way (with the default background),
-     * the method `showModal()` needs to be invoked.
-     *
-     * https://lit.dev/docs/components/lifecycle/#willupdate
-     *
-     * @param changedProperties
-     */
-    willUpdate (changedProperties: Map<string, unknown>) {
-        if (changedProperties.has('isOpen') && this._dialog) {
-            const dialog = this._dialog;
-            const isClosed = changedProperties.get('isOpen') === false;
-
-            if (isClosed) {
-                dialog.showModal();
-            } else {
-                dialog.close();
-            }
-        }
-    }
-
-    _handleCloseDialog = () => {
+    private _handleCloseDialog = () : void => {
```

Comment:
There's two functions here that kinda do the same thing almost (one here and one on line 55), could we merge them so there's one `handleCloseDialog` I think we could move the enable scrolling or keep it in place as long as it's not being called unnecessarily. I would be tempted to delete this function and reuse the one that is on line 55 as they should handle the dispatch event too

Reply:
great spot - updated :)